krun/vm_defs.py

```diff
@@ -110,10 +110,30 @@ def _run_exec(self, args, heap_lim_k, bench_env_changes=None):
         p = subprocess.Popen(
             actual_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
             env=use_env)
-        stdout, stderr = p.communicate()
+
+        stderr_lines = []
+        while True:
+            line = p.stderr.readline()
+
+            if line == "":
+                break  # EOF, child has exited
+
+            line = line.strip()
+            info(line)
+            stderr_lines.append(line)
+
+        stderr = "\n".join(stderr_lines)
+        stdout = p.stdout.read()
+        p.poll()  # sets returncode attr
+
+        # Should not fire. The empty string that causes
+        # the above loop to exit is indicating EOF. Thus the
+        # child should be done.
+        assert p.returncode is not None
```

Comment:
Yes, and it will definitely happen at various points (child processes do all sorts of odd things).